llvm/include/llvm/ProfileData/InstrProfReader.h

```diff
@@ -202,11 +202,13 @@ class InstrProfReader {
   /// instrprof file.
   static Expected<std::unique_ptr<InstrProfReader>>
   create(const Twine &Path, vfs::FileSystem &FS,
-         const InstrProfCorrelator *Correlator = nullptr);
+         const InstrProfCorrelator *Correlator = nullptr,
+         std::optional<std::function<void(Error)>> MaybeWarnFn = {});
```

Comment:
as nice as it is to explicitly state that this can be null, `std::function` can already be null and that's widely used in C++, so I'd just make this a `std::function<>` and use its bool conversion to check if it has a value

Reply:
Done.